go/tools/gazelle/packages/package.go

```diff
@@ -15,7 +15,46 @@ limitations under the License.
 
 package packages
 
-import "strings"
+import (
+	"fmt"
+	"strings"
+)
+
+// PlatformConstraints is a map from config_setting labels (for example,
+// "@io_bazel_rules_go//go/platform:linux_amd64") to a sets of build tags
+// that are true on each platform (for example, "linux,amd64").
+type PlatformConstraints map[string]map[string]bool
+
+// ParsePlatformConstraints parses a string that contains all platforms and
+// tags on each platform. The string must contain an even number of fields,
+// separated by whitespace. Even fields (counting from 0) are platform labels.
```

Comment:
Gazelle needs a table containing `config_setting` labels for each platform and lists of tags that are true for each platform. The tags will usually be GOOS and GOARCH, but there may also be tags for specific environments (for example, `appengine`, `js`) or specific features that would pull in dependencies that shouldn't always be compiled.

We need a simple way to express this table on the command line. There is some redundancy here that could be eliminated by requiring some conventions on `config_setting` naming. Tag lists must be specified though. Two suggestions you mentioned above (as I understand them):

* We could require all `config_setting`s to be in the same package so we only have to write the package name once instead of repeating it for each entry. If users define their own `config_setting`s, they would need to alias or copy the predefined rules if they still wanted to use them.
* We could derive `config_setting` names from the tags. For example, `linux,amd64` becomes `linux_amd64`. This prevents customization of the platform name, but I agree, that's of limited value.

Since the package name and the tag lists still need to be specified, maybe we would end up with two smaller options?

`-platform_package @io_bazel_rules_go//go/platforms -platform_tags darwin,amd64;linux,amd64;windows,amd64`

WDYT?